unit_tests/sources/declarative/file/file_stream_manifest.yaml

```diff
@@ -0,0 +1,190 @@
+      - "%Y-%m-%dT%H:%M:%SZ"
+      - "%Y-%m-%dT%H:%M:%S%z"
+    datetime_format: "%s"
+    cursor_field: "{{ parameters.get('cursor_field',  'updated_at') }}"
+    start_datetime:
+      datetime: "{{ timestamp(config.get('start_date')) | int if config.get('start_date') else day_delta(-730, '%s') }}"
+    start_time_option:
+      inject_into: request_parameter
+      field_name: "{{ parameters['cursor_filter'] }}"
+      type: RequestOption
+
+  base_incremental_stream:
+    $ref: "#/definitions/base_stream"
+    incremental_sync:
+      $ref: "#/definitions/cursor_incremental_sync"
+
+  # Incremental cursor-based streams
+  articles_stream:
+    $ref: "#/definitions/base_incremental_stream"
+    name: "articles"
+    primary_key: "id"
+    schema_loader:
+      type: InlineSchemaLoader
+      schema:
+        type: object
+        $schema: http://json-schema.org/schema#
+        properties:
+          id:
+            type: integer
+        additionalProperties: true
+    incremental_sync:
+      $ref: "#/definitions/cursor_incremental_sync"
+      start_time_option:
+        $ref: "#/definitions/cursor_incremental_sync/start_time_option"
+        field_name: "start_time"
+    retriever:
+      $ref: "#/definitions/retriever"
+      ignore_stream_slicer_parameters_on_paginated_requests: true
+      requester:
+        $ref: "#/definitions/retriever/requester"
+        path: "help_center/incremental/articles"
+      paginator:
+        type: DefaultPaginator
+        pagination_strategy:
+          type: CursorPagination
+          cursor_value: '{{ response.get("next_page", {}) }}'
+          stop_condition: "{{ config.get('ignore_pagination', False) or last_page_size == 0 }}"
+        page_token_option:
+          type: RequestPath
+      record_selector:
+        extractor:
+          type: DpathExtractor
+          field_path: ["articles"]
+
+  article_attachments_stream:
+    $ref: "#/definitions/base_incremental_stream"
+    name: "article_attachments"
+    primary_key: "id"
+    schema_loader:
+      type: InlineSchemaLoader
+      schema:
+        type: object
+        $schema: http://json-schema.org/schema#
+        properties:
+          id:
+            type: integer
+        additionalProperties: true
+    retriever:
+      $ref: "#/definitions/retriever"
+      ignore_stream_slicer_parameters_on_paginated_requests: true
+      requester:
+        $ref: "#/definitions/retriever/requester"
+        path: "help_center/articles/{{ stream_partition.article_id }}/attachments"
+      partition_router:
+        type: SubstreamPartitionRouter
+        parent_stream_configs:
+          - type: ParentStreamConfig
+            parent_key: "id"
+            partition_field: "article_id"
+            stream:
+              $ref: "#/definitions/articles_stream"
+            incremental_dependency: true
+      record_selector:
+        extractor:
+          type: DpathExtractor
+          field_path: ["article_attachments"]
+    file_uploader:
```

Comment:
I added this to the stream level so that it can apply to any type of retrievers. That being said, I'm not sure asyncretrievers, statedelegatingretrievers or propertychunkingretrievers would make a lot of sense here...

Reply:
I actually think it needs to be outside of the retriever. The reason is that if we have client side filtering, we only want to download the files if we know the record will get synced